Tools-Override/ConstructSharedFx.targets

```diff
@@ -0,0 +1,64 @@
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <UsingTask TaskName="GenerateDepsJson" AssemblyFile="$(BuildToolsTaskDir)CoreFx.Tools.dll"/>
+
+  <PropertyGroup>
+    <NewSharedFxVersion>9.9.9</NewSharedFxVersion>
+    <OldSharedFxVersion>1.0.0</OldSharedFxVersion>
+    <TestSharedFxPath>$(TestSharedFxDir)\shared\Microsoft.NETCore.App\$(NewSharedFxVersion)</TestSharedFxPath>
+    <TestSharedFxPathHostFxr>$(TestSharedFxDir)\host\fxr\$(NewSharedFxVersion)</TestSharedFxPathHostFxr>
+    <ToolsDotNetCliSharedFxPath>$(ToolsDir)dotnetcli\shared\Microsoft.NETCore.App\$(OldSharedFxVersion)</ToolsDotNetCliSharedFxPath>
+    <ConstructSharedFxSem>$(ObjDir)constructsharedfxcomplete.sem</ConstructSharedFxSem>
+  </PropertyGroup>
+  
+  <!-- This will generate a version of the dotnet cli that runs with the live
+       build of corefx using the other components gathered from the stable 
+       version consumed from the Tools directory
+       To debug any issues set COREHOST_TRACE=1 and rerun dotnet
+  -->  
+  <Target Name="ConstructTestingHost"
+          BeforeTargets="Build;BuildAllProjects"
+          Condition="!Exists('$(ConstructSharedFxSem)') OR '$(ReconstructSharedFx)' == 'true'" >
+          
+    <ItemGroup>
+      <!-- This will only copy over the related file -->
+      <HostFxrFile Condition="'$(OS)'=='Windows_NT'" Include="$(RuntimePath)hostfxr.dll" />
+      <HostFxrFile Condition="'$(OS)'=='Unix'" Include="$(RuntimePath)libhostfxr.so" />
+      <HostFxrFile Condition="'$(OS)'=='Unix'" Include="$(RuntimePath)libhostfxr.dylib" />
+      <DotnetExe Condition="'$(OS)'=='Windows_NT'" Include="$(RuntimePath)dotnet.exe" />
+      <DotnetExe Condition="'$(OS)'!='Windows_NT'" Include="$(RuntimePath)dotnet" />
+    </ItemGroup>
+    <MakeDir Directories="$(TestSharedFxPathHostFxr)" />
+    <Copy SourceFiles="@(HostFxrFile)" 
+          DestinationFolder="$(TestSharedFxPathHostFxr)" 
+          Condition="Exists('%(RootDir)%(Directory)%(Filename)%(Extension)')" />
+    <Copy SourceFiles="@(DotnetExe)" 
+          DestinationFolder="$(TestSharedFxDir)" 
+          Condition="Exists('%(RootDir)%(Directory)%(Filename)%(Extension)')" />
+    <ItemGroup>
+      <LiveDotNetCliComponents Include="$(RuntimePath)\*.*" Exclude="$(RuntimePath)\xunit.console.netcore.exe" />
+    </ItemGroup>
+    <Copy SourceFiles="@(LiveDotNetCliComponents)" 
```

Comment:
I was imagining the case where you would update the version of the assembly built locally and have to regenerate the deps json anyways. Shouldn't deps.json be updated with every change to product including adding new libraries? You could also pass in ReconstructSharedFx=true to force a new sharedfx for every build.